Sources/TSCUtility/Version.swift

```diff
@@ -45,12 +45,122 @@ public struct Version: Hashable {
     }
 }
 
-extension Version: Comparable {
+/// An error that occurs during the creation of a version.
+public enum VersionError: Error, CustomStringConvertible {
+    /// The version string contains non-ASCII characters.
+    case nonASCIIVersionString(_ versionString: String)
+    /// The version core contains an invalid number of Identifiers.
+    case invalidVersionCoreIdentifiersCount(_ identifiers: [String])
+    /// Some or all of the version core identifiers contain non-numerical characters or are empty.
+    case nonNuumericalOrEmptyVersionCoreIdentifiers(_ identifiers: [String])
+    /// Some or all of the pre-release identifiers contain characters other than alpha-numerics and hyphens.
+    case nonAlphaNuumerHyphenalPrereleaseIdentifiers(_ identifiers: [String])
+    /// Some or all of the build metadata identifiers contain characters other than alpha-numerics and hyphens.
+    case nonAlphaNuumerHyphenalBuildMetadataIdentifiers(_ identifiers: [String])
```

Comment:
I coined the term "alpha-numer-hyphenal" because I can't come up with a succinct description for the character set [0-9A-Za-z-]. I don't like this weird term I made, and would like some help to phrase it better.